src/plugins/dev_tools/public/dev_tools_icon.tsx

```diff
@@ -35,6 +36,20 @@ export function DevToolsIcon({
   title: string;
 }) {
   const [modalVisible, setModalVisible] = useState(false);
+
+  const [devToolTab, setDevToolTab] = useState('');
+
+  const createOpenDevToolAction = createAction<typeof DEVTOOL_OPEN_ACTION>({
```

Comment:
> Just sharing my thought, instead of implementing it with Action/Trigger, I'd simply use a global state for the `modalVisible` and `devToolTab`, perhaps just a `BahaviorSubject`.

sounds like Observable is more lightweight, haven't got a chance to try it.

Reply:
By comparing uiActions and Observable, they are both able to handle events. the uiActions is a central place to manage events and could have context object associate with. It's more powerful and extensible. I could prefer keep the current implementation.